### PR TITLE
Traverse immediate child associations by default

### LINKED ISSUE #22: Proposal for new traversal behavior
The current behavior for `get` is to only show what can be achieved with a single API call, and for `get --traverse` to go all the way up and down the hierarchy. 

Because the data displayed with get is nearly (exactly?) the same as what is returned in a list call, a bare `get` doesn't provide much value beyond what I see in the list (other than doing the grep for me). For some types, like classes, I usually want to see the associated plans so I end up always using `get --traverse` anyway.

```console
$ svc-cat binding get eager-goat-wordpress-mysql-binding
                 NAME                	        SERVICE INSTANCE NAME        	       STATUS
+------------------------------------+-------------------------------------+--------------------+
  eager-goat-wordpress-mysql-binding 	 eager-goat-wordpress-mysql-instance 	 InjectedBindResult
+------------------------------------+-------------------------------------+--------------------+

$ svc-cat binding get eager-goat-wordpress-mysql-binding --traverse
                 NAME                	        SERVICE INSTANCE NAME        	       STATUS
+------------------------------------+-------------------------------------+--------------------+
  eager-goat-wordpress-mysql-binding 	 eager-goat-wordpress-mysql-instance 	 InjectedBindResult
+------------------------------------+-------------------------------------+--------------------+


INSTANCE
                 NAME                 	 SERVICE CLASS NAME 	          SERVICE CLASS UUID          	 SERVICE PLAN NAME 	          SERVICE PLAN UUID           	         STATUS
+-------------------------------------+--------------------+--------------------------------------+-------------------+--------------------------------------+-------------------------+
  eager-goat-wordpress-mysql-instance 	 azure-mysqldb      	 997b8372-8dac-40ac-ae65-758b4a5075a5 	 standard800       	 08e4b43a-36bc-447e-a81f-8202b13e339c 	 ProvisionedSuccessfully
+-------------------------------------+--------------------+--------------------------------------+-------------------+--------------------------------------+-------------------------+


SERVICE CLASS
      NAME      	          DESCRIPTION           	                 UUID
+---------------+--------------------------------+--------------------------------------+
  azure-mysqldb 	 Azure Database for MySQL       	 997b8372-8dac-40ac-ae65-758b4a5075a5
                	 Service
+---------------+--------------------------------+--------------------------------------+


SERVICE PLAN
     NAME     	       DESCRIPTION       	                 UUID                 	 CLASS NAME 	              CLASS UUID
+-------------+-------------------------+--------------------------------------+------------+--------------------------------------+
  standard800 	 Standard Tier, 800 DTUs 	 08e4b43a-36bc-447e-a81f-8202b13e339c 	 asb        	 997b8372-8dac-40ac-ae65-758b4a5075a5
+-------------+-------------------------+--------------------------------------+------------+--------------------------------------+


BROKER
  NAME 	                          URL                          	 STATUS TYPE 	 STATUS VALUE 	 STATUS REASON  	         STATUS MESSAGE
+------+-------------------------------------------------------+-------------+--------------+----------------+--------------------------------+
  asb  	 http://asb-azure-service-broker.asb.svc.cluster.local 	 Ready       	 True         	 FetchedCatalog 	 Successfully fetched catalog
       	                                                       	             	              	                	 entries from broker.
+------+-------------------------------------------------------+-------------+--------------+----------------+--------------------------------+
```

Instead, I propose that we retrieve data based on what we expect to most commonly be seful/sought after. Then keep a full hierarchy traversal behind a flag. Since the flag no longer dictates the # of api calls, and instead limits the amount of information displayed, it could be renamed to `--details`.

```console
$ svc-cat binding get <name>
  Name:           <name>
  Description:    <description>
  UUID:           <uuid>
  Plan:           <name>
  Class:          <name>
  Broker:         <name>

Instances:
+----------+--------------------+------------+
  Name          Description         UUID
+----------+--------------------+------------+
 <name>         <description>       <uuid>
 <name>         <description>       <uuid>

$ svc-cat binding get <name> --details
  Name:           <name>
  Description:    <description>
  UUID:           <uuid>
  Plan:           <name>
  Class:          <name>
  Broker:         <name>

Instances:
+----------+--------------------+------------+
  Name          Description         UUID
+----------+--------------------+------------+
 <name>         <description>       <uuid>
 <name>         <description>       <uuid>

Plan:
  Name:           <name>
  Description:    <description>
  UUID:           <uuid>

Class:
  Name:           <name>
  Description:    <description>
  UUID:           <uuid>

Broker:
  Name:           <name>
  Description:    <description>
  UUID:           <uuid>

$ svc-cat catalog plan get <name>
  Name:           <name>
  Description:    <description>
  UUID:           <uuid>
  Class:          <name>
  Broker:         <name>

Bindings:
+----------+--------------------+------------+
  Name          Description         UUID
+----------+--------------------+------------+
 <name>         <description>       <uuid>
 <name>         <description>       <uuid>

$ svc-cat catalog class get <name>
  Name:           <name>
  Description:    <description>
  UUID:           <uuid>
  Broker:         <name>

Plans:
+----------+--------------------+------------+
  Name          Description         UUID
+----------+--------------------+------------+
 <name>         <description>       <uuid>
 <name>         <description>       <uuid>
```

### PR DESCRIPTION
**Pull request by carolynvs**
Continuing from #43.
___

- [x] `svc-cat describe instance <name>` prints child bindings by default
- [x] `svc-cat describe plan <name>` shouldn't hide the instances behind `--traverse` and instead should print the parent class and broker.
- [x] `svc-cat describe class <name>` shouldn't hide the plans behind `--traverse` and should print the parent broker.

Closes #22.

### CODE REVIEW

File: pkg/instance/describe_cmd.go [hunk 1]
```diff
@@ -63,6 +63,12 @@ func (c *describeCmd) describe(name string) error {
 
 	output.WriteInstanceDetails(instance)
 
+	bindings, err := traverse.InstanceToBindings(c.cl, instance)
```
Comment (arschles) [inline pkg/instance/describe_cmd.go]: @carolynvs should we have a way to opt out of this? I didn't see one

File: pkg/plan/describe_cmd.go [hunk 2]
```diff
@@ -71,22 +70,26 @@ func (c *describeCmd) describe(key string) error {
...
-	class, err := c.cl.ServicecatalogV1beta1().ClusterServiceClasses().Get(plan.Spec.ClusterServiceClassRef.Name, v1.GetOptions{})
+	class, err := traverse.PlanToClass(c.cl, plan)
 	if err != nil {
-		return fmt.Errorf("unable to get class (%s)", err)
+		return err
 	}
 
 	output.WritePlanDetails(plan, class)
 
+	instances, err := traverse.PlanToInstances(c.cl, plan)
```
Comment (arschles) [inline pkg/plan/describe_cmd.go]: same here RE opting out of this functionality

**Comment by carolynvs** [comment]
@arschles I thought I implemented what we discussed in #22, but maybe we both came away with different ideas? 😀 

Currently get and describe show nearly the same information (see below). My intent with that issue was to retrieve useful child relationships by default, like from a class to its plans, and that `--traverse` would be used to display the parent hierarchy.

It sounds like you are asking to have the child relationships also put behind the `--traverse` flag? If not, could you provide sample command + output to help clarify?
```console
$ svcat get class azure-mysqldb
      NAME                 DESCRIPTION                             UUID
+---------------+--------------------------------+--------------------------------------+
  azure-mysqldb   Azure Database for MySQL         997b8372-8dac-40ac-ae65-758b4a5075a5
                  (Experimental)

$ svcat describe class azure-mysqldb
  Name:          azure-mysqldb
  Description:   Azure Database for MySQL (Experimental)
  UUID:          997b8372-8dac-40ac-ae65-758b4a5075a5
  Status:        Active
  Tags:          Azure, MySQL, Database
```

**Comment by arschles** [comment]
@carolynvs I recall that discussion now, and you're right. Sorry about that! This LGTM

**Pull request merged**